iib/workers/tasks/build_add_deprecations.py

```diff
@@ -0,0 +1,198 @@
+    attributes=get_binary_versions(),
+)
+def handle_add_deprecations_request(
+    request_id: int,
+    operator_package: str,
+    deprecation_schema: str,
+    from_index: Optional[str] = None,
+    binary_image: Optional[str] = None,
+    overwrite_from_index: bool = False,
+    overwrite_from_index_token: Optional[str] = None,
+    build_tags: Optional[Set[str]] = None,
+    binary_image_config: Optional[Dict[str, Dict[str, str]]] = None,
+) -> None:
+    """
+    Add a deprecation schema to index image.
+
+    :param int request_id: the ID of the IIB build request
+    :param str operator_package: Operator package of deprecation schema
+    :param str deprecation_schema: deprecation_schema to be added to index image
+    :param str from_index: the pull specification of the container image containing the index that
+        the index image build will be based from.
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from.
+    :param bool overwrite_from_index: if True, overwrite the input ``from_index`` with the built
+        index image.
+    :param str overwrite_from_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required to use ``overwrite_from_index``.
+        The format of the token must be in the format "user:password".
+    :param list build_tags: List of tags which will be applied to intermediate index images.
+    :param dict binary_image_config: the dict of config required to identify the appropriate
+        ``binary_image`` to use.
+    """
+    _cleanup()
+    set_request_state(request_id, 'in_progress', 'Resolving the index images')
+
+    prebuild_info = prepare_request_for_build(
+        request_id,
+        RequestConfigAddDeprecations(
+            _binary_image=binary_image,
+            from_index=from_index,
+            overwrite_from_index_token=overwrite_from_index_token,
+            operator_package=operator_package,
+            deprecation_schema=deprecation_schema,
+            binary_image_config=binary_image_config,
+        ),
+    )
+
+    from_index_resolved = prebuild_info['from_index_resolved']
+    binary_image_resolved = prebuild_info['binary_image_resolved']
+    Opm.set_opm_version(from_index_resolved)
+
+    _update_index_image_build_state(request_id, prebuild_info)
+
+    with tempfile.TemporaryDirectory(prefix=f'iib-{request_id}-') as temp_dir:
+        operators_in_db, index_db_path = verify_operators_exists(
+            from_index_resolved,
+            temp_dir,
+            [operator_package],
+            overwrite_from_index_token,
+        )
+        if not operators_in_db:
+            log.debug(
+                '%s operator package is either not present in index or it is opted in fbc',
+                operator_package,
+            )
+            err_msg = f'Cannot add deprecations for {operator_package}'
+            log.error(err_msg)
```

Comment:
Also, we probably don't need both `log.debug` and `log.error`. Simply logging the error should be good, no?